dipy/reconst/dti.py

```diff
@@ -777,17 +782,35 @@ def fit(self, data, mask=None):
             min_signal = self.min_signal
 
         data_in_mask = np.maximum(data_in_mask, min_signal)
-        params_in_mask = self.fit_method(self.design_matrix, data_in_mask,
-                                         *self.args, **self.kwargs)
+
+        if self.return_S0_hat:
```

Comment:
Yes - that's cleaner. In general, I prefer it when there is only one call to the function. That way, if the API changes for reasons unrelated to `S0_hat`, we only have to change the call in one place, instead of two places.